contracts/test/EVMCompatabilityTest.js

```diff
@@ -1072,6 +1073,34 @@ describe("EVM Test", function () {
   });
 });
 
+describe("Ethereum Validations ", function() {
+  it("should prevent wrong chainId", async function() {
+    await setupSigners(await hre.ethers.getSigners())
+    const signer = generateWallet()
+    await fundAddress(await signer.getAddress())
+
+    // const accounts = await setupSigners(await hre.ethers.getSigners())
+    const signedTx = await signer.signTransaction({
+      to: await signer.getAddress(),
+      value: 0,
+      chainId: "0x530",
```

Comment:
maybe we shouldnt use a chain id of one of our chains in case we point it at atlantic2